ipa-core/src/net/client/mod.rs

```diff
@@ -274,22 +287,44 @@ impl MpcHelperClient {
         }
     }
 
+    /// Helper to read a possible error response to a request that returns nothing on success
+    ///
+    /// # Errors
+    /// If there was an error reading the response body or if the request itself failed.
+    pub async fn resp_ok(resp: ResponseFromEndpoint<'_>) -> Result<(), Error> {
+        if resp.status().is_success() {
+            Ok(())
+        } else {
+            Err(Error::from_failed_resp(resp).await)
+        }
+    }
+
+    /// Reads the entire response from the server into a String
+    ///
+    /// # Errors
+    /// If there was an error collecting the response stream.
+    async fn response_to_str(resp: ResponseFromEndpoint<'_>) -> Result<String, Error> {
```

Comment:
you shouldn't require strings to deal with serialization, serde_json can use byte slices for deserialization. This will make your code easier as you can use `to_bytes` on `Aggregated`